### PR TITLE
CI: use codecov/codecov-action

### PR DESCRIPTION
**Pull request by k-takata**
Use codecov/codecov-action instead of the codecov CLI.

### CODE REVIEW

**Review by masatake — APPROVED** [review]

**Comment by masatake** [comment]
Thank you very much.

File: .github/workflows/testing-gnulinux.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       # Following cases target ctags without gcov.
       SANDBOX_CASES: sandbox,sandbox-crash,sandbox-default-req,sandbox-unknown-submode
       DEBIAN_FRONTEND: noninteractive
-      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+      #CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (k-takata) [inline .github/workflows/testing-gnulinux.yml:28]: Is it better to remove this line?

File: .github/workflows/testing-gnulinux.yml [hunk 2]
```diff
@@ -117,4 +117,12 @@ jobs:
         # lcoveralls --root . --retry-count 5 ${BUILDDIR}/coverage.info
 
         # Upload to codecov.
-        bash <(curl -s https://codecov.io/bash) -f ${BUILDDIR}/coverage.info -Z -X gcov -X coveragepy -X search -X xcode -X gcovout -n travis-${{ matrix.os }}
+        #bash <(curl -s https://codecov.io/bash) -f ${BUILDDIR}/coverage.info -Z -X gcov -X coveragepy -X search -X xcode -X gcovout -n travis-${{ matrix.os }}
```
Comment (k-takata) [inline .github/workflows/testing-gnulinux.yml:120]: Is it better to remove these lines?

Also L116-117 can be removed?

**Comment by k-takata** [comment]
I tried to remove lcov, but it didn't work well. It seems that lcov is still needed even with codecov/codecov-action.

**Comment by masatake** [comment]
Thank you again.

**Pull request merged**